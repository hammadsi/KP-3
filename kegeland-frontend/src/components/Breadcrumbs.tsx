import {
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  Text,
} from '@chakra-ui/react';

import { RoutePath } from '../routes';

type BreadcrumbsProps = {
  crumbs: RoutePath[];
} & React.ComponentProps<typeof Breadcrumb>;

const Breadcrumbs: React.FC<BreadcrumbsProps> = ({ crumbs, ...props }) => {
  return (
    <Breadcrumb {...props}>
      <BreadcrumbItem>
        <BreadcrumbLink href="/">
          <Text color="gray.700">Home</Text>
        </BreadcrumbLink>
      </BreadcrumbItem>
      {crumbs.map((crumb) => (
        <BreadcrumbItem key={crumb.match.pattern.path}>
          <BreadcrumbLink
            href={crumb.match.pathname}
            color={crumb.active ? 'primary.600' : 'gray.700'}>
            {crumb.active ? (
              <Text noOfLines={1} fontWeight="medium">
                {crumb.title}
              </Text>
            ) : (
              <Text
                noOfLines={1}
                maxW="40"
                width="full"
                wordBreak="break-all"
                fontWeight="normal">
                {crumb.title}
              </Text>
            )}
          </BreadcrumbLink>
        </BreadcrumbItem>
      ))}
    </Breadcrumb>
  );
};

export default Breadcrumbs;
