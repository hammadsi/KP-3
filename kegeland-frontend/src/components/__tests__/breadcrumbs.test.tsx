import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import {
  routePathLoginFalseMock,
  routePathPatientPatientIdMock,
} from '../../utils/mocks/breadcrumb.mock';
import Breadcrumbs from '../Breadcrumbs';

describe('Test bredcrumb', () => {
  it('renders correctly empty', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Breadcrumbs crumbs={[]} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly patient', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Breadcrumbs crumbs={[routePathPatientPatientIdMock]} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly lgoin false', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Breadcrumbs crumbs={[routePathLoginFalseMock]} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
