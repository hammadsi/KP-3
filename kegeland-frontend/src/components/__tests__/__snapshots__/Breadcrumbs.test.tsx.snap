// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test bredcrumb renders correctly empty 1`] = `
<nav
  aria-label="breadcrumb"
  className="chakra-breadcrumb css-0"
>
  <ol
    className="chakra-breadcrumb__list css-70qvj9"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="/"
      >
        <p
          className="chakra-text css-735bf1"
        >
          Home
        </p>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Test bredcrumb renders correctly lgoin false 1`] = `
<nav
  aria-label="breadcrumb"
  className="chakra-breadcrumb css-0"
>
  <ol
    className="chakra-breadcrumb__list css-70qvj9"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="/"
      >
        <p
          className="chakra-text css-735bf1"
        >
          Home
        </p>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-735bf1"
        href="/login"
      >
        <p
          className="chakra-text css-hs9m67"
        >
          Login
        </p>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Test bredcrumb renders correctly patient 1`] = `
<nav
  aria-label="breadcrumb"
  className="chakra-breadcrumb css-0"
>
  <ol
    className="chakra-breadcrumb__list css-70qvj9"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="/"
      >
        <p
          className="chakra-text css-735bf1"
        >
          Home
        </p>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-lalzaa"
        href="/patients/patientId"
      >
        <p
          className="chakra-text css-1wfrv4i"
        >
          Patient-patientId
        </p>
      </a>
    </li>
  </ol>
</nav>
`;
